module/VuFind/src/VuFind/Http/GuzzleService.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * Guzzle service.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development
+ */
+
+namespace VuFind\Http;
+
+/**
+ * Guzzle service.
+ *
+ * N.B. Use only for dependencies that require Guzzle.
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development
+ * @todo     Merge with PSR-18 HTTP Client Service when implemented
+ */
+class GuzzleService
+{
+    /**
+     * Default regular expression matching a request to localhost.
+     *
+     * @var string
+     */
+    public const LOCAL_ADDRESS_RE = '@^(localhost|127(\.\d+){3}|\[::1\])@';
+
+    /**
+     * VuFind configuration
+     *
+     * @var array
+     */
+    protected $config;
+
+    /**
+     * Regular expression matching a request to localhost or hosts
+     * that are not proxied.
+     *
+     * @see \Laminas\Http\Client\Adapter\Proxy::$config
+     *
+     * @var string
+     */
+    protected $localAddressesRegEx = self::LOCAL_ADDRESS_RE;
+
+    /**
+     * Mappings from VuFind HTTP settings to Guzzle
+     *
+     * @var array
+     */
+    protected $guzzleHttpSettingsMap = [
+        'sslcafile' => 'ssl_key',
```

Comment:
Is there a mapping for sslcapath? At least in some situations, that setting is required on Debian-based systems.

Reply:
It was missing and sslcafile mapping was invalid. Should be fixed now.